src/test/java/warehouse/CartTest.java

```diff
@@ -33,6 +33,31 @@ void getTotalPrice() {
     }
 
     @Test
-    void addToCartById() {
+    void addToCartById_cartValidSize() throws InvalidIdException, EmptyWarehouseException {
+        Product product1 = new Product(DeviceType.SMARTPHONE, "Apple", "xs", "ok", 20, 180, 1200, 1250, 1382);
+        warehouse.addProductToStock(product1);
+        cart.addToCartById(new FilterById(1), warehouse);
+        assertEquals(1, cart.getDevices().size());
+    }
+
+    @Test
+    void addToCartById_cartContainsProducts() throws InvalidIdException, EmptyWarehouseException {
+        Product product1 = new Product(DeviceType.SMARTPHONE, "Apple", "xs", "ok", 10, 180, 1200, 1250, 1382);
+        Product product2 = new Product(DeviceType.SMARTPHONE, "Apple1", "xs", "ok", 10, 180, 1200, 1250, 1382);
+        Product product3 = new Product(DeviceType.SMARTPHONE, "Apple2", "xs", "ok", 10, 180, 1200, 1250, 1382);
+        warehouse.addProductToStock(product1);
+        warehouse.addProductToStock(product2);
+        warehouse.addProductToStock(product3);
+        cart.addToCartById(new FilterById(1), warehouse);
+        cart.addToCartById(new FilterById(2), warehouse);
+        cart.addToCartById(new FilterById(3), warehouse);
+        assertTrue(cart.getDevices().contains(product1));
+    }
+
+    @Test
+    void addToCartById_warehouseIsEmpty() throws InvalidIdException, EmptyWarehouseException {
```

Comment:
dai direttamente il nome dell'exeption dato che la stai testando;
ad esempio : "CheckInvalidIdExceptionWhenTheWarehouseIsEmpty"
e cancella il il warehouse is empty a linea 60 che non serve a nulla